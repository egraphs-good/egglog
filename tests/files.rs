use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
