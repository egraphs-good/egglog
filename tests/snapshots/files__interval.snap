---
source: tests/files.rs
expression: serialized
---
nodes:
  Rational-0:
    op: (rational -10 1)
    children: []
    eclass: Rational-0
    cost: 0
  lo-0:
    op: lo
    children:
      - Var-7848520443469635519
    eclass: Rational-0
    cost: 1
  Rational-11743562013128004906:
    op: (rational 100 1)
    children: []
    eclass: Rational-11743562013128004906
    cost: 0
  lo-5871781006564002453:
    op: lo
    children:
      - e-0
    eclass: Rational-11743562013128004906
    cost: 1
  Mul-0:
    op: Mul
    children:
      - x-0
      - Var-7848520443469635519
    eclass: "1"
    cost: 1
  x-0:
    op: x
    children: []
    eclass: "0"
    cost: 9223372036854776000
  String-7848520443469635519:
    op: "\"x\""
    children: []
    eclass: String-7848520443469635519
    cost: 0
  Var-7848520443469635519:
    op: Var
    children:
      - String-7848520443469635519
    eclass: "0"
    cost: 1
  e-0:
    op: e
    children: []
    eclass: "1"
    cost: 9223372036854776000
  Rational-5871781006564002453:
    op: (rational 10 1)
    children: []
    eclass: Rational-5871781006564002453
    cost: 0
  hi-0:
    op: hi
    children:
      - x-0
    eclass: Rational-5871781006564002453
    cost: 1
root_eclasses: []
class_data:
  Rational-0:
    type: Rational
  "0":
    type: Math
  Rational-11743562013128004906:
    type: Rational
  "1":
    type: Math
  String-7848520443469635519:
    type: String
  Rational-5871781006564002453:
    type: Rational

